Sources/NIO/Socket.swift

```diff
@@ -20,7 +20,14 @@ public typealias IOVector = iovec
 
     /// The maximum number of bytes to write per `writev` call.
     static var writevLimitBytes: Int {
-        return Int(Int32.max)
+        #if arch(arm) // 32-bit, Raspi/AppleWatch/etc
+            // Note(hh): This is not a _proper_ fix, but necessary because
+            //           other places extend on that. Should be fine in
+            //           practice on 32-bit platforms.
+            return Int(Int32.max / 4)
```

Comment:
Hrm, I don't love this. AFAICT this change works around the fact that we have a test that validates what happens if we attempt to write more than `writevLimitBytes`, which does some unguarded addition on this value.

For the test in `ChannelTests.swift`, it may be best to change it on 32-bit systems to create the buffer at `writevLimitBytes` and adjust the expected return values. That's the largest possible allocated buffer on 32-bit NIO anyway.

Reply:
No, there are more NIO issues wrt that. I.e. ByteBuffer uses UInt32 and nextPoW2 to round up the size, which subsequently fails when casting backt to malloc, which wants an Int.
It simply is no good to live on that edge on 32-bit platforms. IMO it is not worthwhile to adjust all of NIO just for the 32-bit edge case. And you don't do similar edge-checks for 64-bit either (but stick to Int32.max ;-) )

Reply:
> ByteBuffer uses UInt32 and nextPoW2 to round up the size, which subsequently fails when casting backt to malloc

Doesn't this patch change that?

Reply:
I tracked down this specific thing, yes (but stuff kept failing). What I was trying to say is that it probably makes no sense to support that max-addressable-memory edge-case in NIO (generally). If you *really* want to, I suggest starting out by writing tests that check against Int64.max on 64-bit too. It is not worth it, IMO.

Reply:
@helje5 I don't understand why `Int32.max` doesn't work here. It fits into an `Int` on both 32 and 64 bit platforms.

Reply:
```
            // Note(hh): This is not a _proper_ fix, but necessary because
            //           other places extend on that. Should be fine in
            //           practice on 32-bit platforms.
```
Well, it is in the comment above it (and in my reply above yours). It is not a proper fix, but it is a lot of work to track down all edge cases (for little gain).

P.S.: *Maybe* a lot of that went away when you changed sysMalloc(Int) to sysMalloc(size_t).

Reply:
@helje5 could we re-check this?

Reply:
@helje5 if you could just re-check that this is still the case. And then just remove the two `note(hh)` comments with a good explanation why we need (if we need) special casing.

Reply:
if that's done we can merge I think.